imports/api/classes/Server.jsx

```diff
@@ -686,28 +718,53 @@ export default class Server {
     }
 
     smtpReceive(file) {
-        const that = this;
         const mms = MM4.parseMms(file);
+        const msgId = mms.messageId;
+        let message = MessageDB.findOne({message_id: msgId});
+        let owner = this.didOwner(message.from);
         if (MM4.isResponse(mms.messageType)) {
             showDebug('MMS send server response: %s', mms);
-            //this confirms that mms send is successful
-
-            //TODO:
-            //get related document from mongo db (use manager class when updating record)
-            //set result object
-            //set message id
-            //update document
+            const Message = new MessageManager(owner.account_id);
+            Message.parseJSON(message);
+            Message.parseJSON(mms);
```

Comment:
Why parse twice?

Reply:
this.json._id 

or should I edit the update to use message_id ?

Reply:
will remove pala Message.parseJSON(mms); since it's the same sorry